lambdas/get-addresses/tests/unit/app.test.ts

```diff
@@ -0,0 +1,72 @@
+import { APIGatewayProxyEvent } from "aws-lambda";
+import { lambdaHandler } from "../../src/app";
+import { DynamoDbClient } from "../../src/lib/dynamo-db-client";
+import { CanonicalAddress } from "../../src/types/address";
+
+const mockDynamoDbClient = jest.mocked(DynamoDbClient);
+
+describe("Handler", () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it("should return an address when an address is found", async () => {
+        const addresses: CanonicalAddress[] = [
+            {
+                streetName: "Downing street",
+                buildingNumber: "10",
+                postalCode: "SW1A 2AA",
+            },
+        ];
+
+        const savedAddress = { Item: addresses };
+
+        mockDynamoDbClient.send = jest.fn().mockResolvedValue(savedAddress);
+
+        const params = {
+            headers: {
+                session_id: "123-abc",
+            },
+        } as unknown as APIGatewayProxyEvent;
+
+        const result = await lambdaHandler(params);
+        const resultAddress = JSON.parse(result.body).result;
+        expect(resultAddress).toEqual(addresses);
```

Comment:
Can also do an assert for `sessionId` returned in the payload for this and other tests?
E.g.
`expect(resultAddress.sessionId).toEqual('123-abc');`

Reply:
I'm not sure that would be needed as we're testing the `session_id` exists in the `GetCommand` in the `address-service.test.ts`. We also dont use the sessionId after its returned so we would be testing something not needed (imo)